### PR TITLE
dark mode toggle

### PR DESCRIPTION
**Pull request by ltclmbdev**
Apply dark-mode here, reworked theme (farther clean up needed though).
Also did some refactor across the app (mostly cleanup the styles).

<details>
  <summary>Home page default:</summary>
  
</details>
<br/>
<details>
  <summary>Home page dark:</summary>
  
</details>
<br/>
<details>
  <summary>Posts page default:</summary>
  
</details>
<br/>
<details>
  <summary>Posts page dark:</summary>
  
</details>
<br/>
<details>
  <summary>Post page default:</summary>
  
</details>
<br/>
<details>
  <summary>Post page dark:</summary>
  
</details>
<br/>
<br/>

200w_d

### CODE REVIEW

**Comment by theianjones** [comment]
Could we make these cards some share of gray? At least when I think of dark mode, I would be surprised to see that much white.

**Comment by vojtaholik** [comment]
> Could we make these cards some share of gray? At least when I think of dark mode, I would be surprised to see that much white. 

Yeah, should be same color as the background. @DrShpongle check out Joel's blog.

**Comment by vojtaholik** [comment]
Why is it pink? Could just reverse the colors.

**Comment by ltclmbdev** [comment]
I set up the theme toggling behavior, colors were set up approximately. I
took colors from Joel's blog. Will fix it 👍

**Comment by ltclmbdev** [comment]
Updated the colors (and screenshots in description). Hope this variont is more "darkish" 🕶

File: gatsby-node.js [hunk 1]
```diff
@@ -84,6 +84,7 @@ exports.onCreateWebpackConfig = ({ actions }) => {
     resolve: {
       modules: [path.resolve(__dirname, 'src'), 'node_modules'],
       alias: {
+        'react-dom': '@hot-loader/react-dom',
```
Comment (ltclmbdev) [inline gatsby-node.js:87]: it treats this warning:
Image 2019-06-02 at 12 07 35 PM

from here: https://github.com/gaearon/react-hot-loader/issues/1227#issuecomment-482139583

**Review by joelhooks — APPROVED** [review]

**Pull request merged**